presto-orc/src/main/java/com/facebook/presto/orc/reader/LongSelectiveStreamReader.java

```diff
@@ -73,12 +74,20 @@ public void startStripe(Stripe stripe)
                     directReader = new LongDirectSelectiveStreamReader(context);
                 }
                 currentReader = directReader;
+                if (dictionaryReader != null && context.isResetAllReaders()) {
+                    dictionaryReader.startStripe(stripe);
+                    System.setProperty("RESET_LONG_READER", "RESET_LONG_READER");
+                }
                 break;
             case DICTIONARY:
                 if (dictionaryReader == null) {
                     dictionaryReader = new LongDictionarySelectiveStreamReader(context);
                 }
                 currentReader = dictionaryReader;
+                if (directReader != null && context.isResetAllReaders()) {
+                    directReader.startStripe(stripe);
+                    System.setProperty("RESET_LONG_READER", "RESET_LONG_READER");
```

Comment:
Don't set system properties in OSS, remove this from all classes. If you need it for testing use a build with local changes.